File: 1894-merge-strings-alternately/1894-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int maxLength = max(word1.size(), word2.size());
        string res;
        for(int i = 0; i < maxLength; i++){
            if(i + 1 <= word1.size()){
                res.push_back(word1[i]);
            }

            if(i + 1 <= word2.size()){
                res.push_back(word2[i]);
            }
        }

        return res;
    }
};

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        i = 0
        j = 0
        n = len(word1)
        m = len(word2)
        res = []
        while i < n and j < m:
            res.append(word1[i])
            res.append(word2[j])
            i +=1
            j +=1
        res.append(word1[i:n])
        res.append(word2[j:m])
        return "".join(res)